dot/parachain/types/descriptor_v2.go

```diff
@@ -0,0 +1,226 @@
+package parachaintypes
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+type CandidateDescriptorVersion byte
+
+const (
+	// // The old candidate descriptor version.
+	CandidateDescriptorVersion1 CandidateDescriptorVersion = iota + 1
+
+	// The new candidate descriptor version.
+	CandidateDescriptorVersion2
+)
+
+var ErrUnknownCandidateDescriptorVersion = fmt.Errorf("unknown candidate descriptor version")
+
+// CandidateDescriptorV2 is a descriptor for a parachain candidate.
+type CandidateDescriptorV2 struct {
```

Comment:
just moved from types.go to this file. this is not the new type added in this PR.